test/parallel/test-vm-module-basic.js

```diff
@@ -107,3 +112,38 @@ const util = require('util');
   assert.notStrictEqual(dep, undefined);
   assert.strictEqual(dep, m.dependencySpecifiers);
 }
+
+// Check the impossibility of creating an abstract instance of the Module.
+{
+  common.expectsError(() => new Module(), {
+    message: 'Module is not a constructor',
+    type: TypeError
+  });
+}
+
+// Check to throws invalid exportNames
+{
+  common.expectsError(() => new SyntheticModule(undefined, () => {}, {}), {
+    message: 'The "exportNames" argument must be of type Array of strings.' +
+      ' Received type undefined',
+    type: TypeError
+  });
+}
+
+// Check to throws invalid evaluateCallback
+{
+  common.expectsError(() => new SyntheticModule([], undefined, {}), {
+    message: 'The "evaluateCallback" argument must be of type function.' +
+      ' Received type undefined',
+    type: TypeError
+  });
+}
+
+// Check to throws invalid options
+{
+  common.expectsError(() => new SyntheticModule([], () => {}, null), {
+    message: 'The "options" argument must be of type Object.' +
+      ' Received type object',
```

Comment:
Not related to this PR but this error message is confusing.

Reply:
This is fixed by https://github.com/nodejs/node/pull/29675